docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,514 @@
+
+> ❗️ The `process.env.SITE_URL` variable has to be correctly configured and accessible from both server and client side, otherwise the request to the proxy endpoint will not happen correctly. This may mean that preview url's in merge request will not function correctly, which will happen if the application is started with `process.env.NODE_ENV` set to `production`.
+
+## Custom SEO
+
+There may be cases where you will want to configure custom SEO per page. Unfortunately, Contentful does not provide out-of-the-box SEO support, so you will need to implement your own strategy. The approach we suggest is the following:
+
+- Create a content model for SEO.
+- Add a field `Title` that should be only used to identify the model, e.g. "Homepage SEO".
+- Add a SEO field (not-rich long text) to the model. Here, the SEO related tags (title, meta, etc.) will be defined in a _json_ format like so:
+
+```json
+{
+    "title": "MyPage Title",
+    "meta": [
+        {
+            "name": "description",
+            "content":"MyPage Description",
+        },
+        {
+            "property": "og:title",
+            "content": "MyPage Title",
+        },
+        {
+            "property": "og:image",
+            "content": {
+                "id": "6fU8dkL1P9eZlPE9JPw89n"
+            }
+        }
+    ],
+}
+```
+
+- Add a SEO Assets field (many file, media input) to the model.
+- Create a link field type in the models you need SEO and link it to the model you just created.
+
+The resulting content model should look like this.
+
+![SEO Content Model](../../static/img/SEO%20content%20model.png)
+
+In order to obtain the `id` used for meta tags whose content is an asset, you need to select the asset entry and, on the right side panel of the asset, select **Info** and copy the **id**. Later in the application you'll use this **id** to generate the URL for the asset.
+
+On the application itself, the steps to customize the SEO are the following:
+
+- Define a _default_ SEO data.
+- Fetch page data as you normally would.
+- Obtain the SEO data from the Contentful page data (example code for the Contentful SEO _parser_)
+
+```js
+import { mapValues } from 'lodash';
+
+// This function will determine if a meta entry's content is an id
+// If it is, it obtains the url to the asset that matches the id
+// Otherwise it returns the content 
+
+const getContent = (content, assets) => {
+    if (isPlainObject(content)) {
+        const asset = assets.find(({ sys }) => sys.id === content.id);
+
+        return asset.fields.file.url.slice(2);
+    }
+
+    return content;
+};
+
+// Provided with the result from the Contentful API call and the entryID for the current page content,
+// This function will extract the SEO data, and complete it with the URL of any needed asset.
+
+const contentfulSEOParser = (contentfulData, entryID) => {
```

Comment:
Can we name it `parseContentfulSEO` or something similar?

Reply:
Yeah, I think it's a good idea to have it similar in both recipes 👍